crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -857,8 +858,11 @@ fn is_error_retryable(error: &SuiError) -> bool {
         SuiError::ExecutionError(..) => false,
         SuiError::ByzantineAuthoritySuspicion { .. } => false,
         SuiError::QuorumFailedToGetEffectsQuorumWhenProcessingTransaction { .. } => false,
+        SuiError::ObjectVersionUnavailableForConsumption { .. } => false,
```

Comment:
Wouldn't this be retry-able if the validator is simply missing the parent transaction?

Reply:
If we only look at one validator, yes. But we consider a tx non-retryable only if there are >= f+1 non retryable errors. We shouldn't have f+1 validators speak `ObjectVersionUnavailableForConsumption`

Reply:
sorry, I was confused. So `ObjectVersionUnavailableForConsumption` means the given object version is older than the  currently locked object version. When we see this error,, it means means the transaction is invalid, mostly because a certain fullnode tells the user to use a stale object.
I think you were referring to `TransactionInputObjectsErrors` which would include `ObjectNotFound` error or a similar one that signals the validators that haven't executed the tx that leads to the input transaction object(s). However it needs some modification as well because now we treat it as non-retryable. Will fix it in another PR